### PR TITLE
Implemented the Storage

### PR DESCRIPTION
**Pull request by Viktoriia-Zinchenko**

### CODE REVIEW

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 1]
```diff
@@ -3,17 +3,40 @@
...
+    private final V[] values;
+    private int amount;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[STORAGE_SIZE];
+        values = (V[]) new Object[STORAGE_SIZE];
+    }
+
     @Override
+
```
Comment (karma-o) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:17]: This line is redundant

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 2]
```diff
@@ -3,17 +3,40 @@
...
     public void put(K key, V value) {
+        if (get(key) == null) {
```
Comment (karma-o) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:19]: You are checking only if it is `null`, but how do we check if element with such a `key` already exists,
for example if `key == null` and `value != null`, is it just overwrites the value if it sees `null`?
I think a condition like this:
`key == keys[i] || key != null && key.equals(keys[i]`
would be better, and more readable, but I am not sure, we better ask a mentor about this.

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 3]
```diff
@@ -3,17 +3,40 @@
...
+            keys[amount] = key;
+            values[amount] = value;
+            amount++;
+        } else {
+            values[amount - 1] = value;
+        }
```
Comment (karma-o) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:25]: I think `else` with its braces is redundant and we can remove it.

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 4]
```diff
@@ -3,17 +3,40 @@
...
     }
 
     @Override
     public V get(K key) {
+        for (int i = 0; i < amount; i++) {
```
Comment (karma-o) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:30]: Smart move with looping only by dynamically changing `amount` of storage, and not by all `STORAGE_SIZE` 🔥

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 5]
```diff
@@ -3,17 +3,39 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+  private static final int STORAGE_SIZE = 10;
+    private final K[] keys;
```
Comment (AlexRozv) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:7]:
```suggestion
    private static final int STORAGE_SIZE = 10;
    private final K[] keys;
```

Comment (AlexRozv) [reply src/main/java/core/basesyntax/impl/StorageImpl.java:7]: let's fix identation

**Review by AlexRozv — CHANGES_REQUESTED** [review]

**Review by AlexRozv — APPROVED** [review]
good job 👍

**Pull request closed without merging**